### ISSUE TITLE
unnecessary qualification error on impl async fn

### ISSUE THREAD

**Issue by beckend** [rust-lang/rust#122533]
I tried this code:

```rust
use async_trait;

#[async_trait::async_trait]
pub trait DropAsync {
  /// The method needed to drop.
  async fn async_drop(&mut self);
}

#[derive(Default)]
struct TestDropper<'a> {
  callback: Option<Box<dyn FnOnce() + Send + 'a>>,
}

#[async_trait::async_trait]
impl DropAsync for TestDropper<'static>
where
  Self: Send,
{
  async fn async_drop(&mut self) {
    todo!();
  }
}
```

I expected to see this happen: No errors.

Instead, this happened: rustc compile error. Both on trait definition and impl block.

### Meta

`rustc --version --verbose`:
```
rustc 1.78.0-nightly (3cbb93223 2024-03-13)
binary: rustc
commit-hash: 3cbb93223f33024db464a4df27a13c7cce870173
commit-date: 2024-03-13
host: x86_64-unknown-linux-gnu
release: 1.78.0-nightly
LLVM version: 18.1.0
```

<details><summary>Backtrace</summary>
<p>

```
error: unnecessary qualification
   |
   |   async fn async_drop(&mut self) {
   |   ^^^^^
```

</p>
</details>

**Comment by beckend** [comment]
Ok, I will try again locally thanks.

**Comment by Edgeworth** [comment]
This has been happening to me for a while, currently on rustc 1.78.0-nightly (c67326b06 2024-03-15)

**Comment by dtantsur** [comment]
I also definitely see absurd results from this lint, examples: https://github.com/dtantsur/rust-openstack/actions/runs/8308800477/job/22739423851?pr=153 (some are kinda valid, ignore them).

**Comment by beckend** [comment]
Ok reopening.

**Comment by Edgeworth** [comment]
https://github.com/dtolnay/async-trait/issues/259

Think this will be fixed now though?
